pkg/shuffle/shuffle.go

```diff
@@ -0,0 +1,44 @@
+package shuffle
+
+import (
+	"github.com/numaproj/numaflow/pkg/isb"
+	"hash"
+	"hash/fnv"
+)
+
+// Shuffle shuffles messages among ISB
+type Shuffle struct {
+	bufferIdentifiers []string
+	buffersCount      int
+	hash              hash.Hash64
+}
+
+// NewShuffle accepts list of buffer identifiers(unique identifier of isb)
+// and returns new shuffle instance
+func NewShuffle(bufferIdentifiers []string) *Shuffle {
+	return &Shuffle{
+		bufferIdentifiers: bufferIdentifiers,
+		buffersCount:      len(bufferIdentifiers),
+		hash:              fnv.New64(),
+	}
+}
+
+// ShuffleMessages accepts list of isb messages and returns the mapping of isb to messages
+func (s *Shuffle) ShuffleMessages(messages []*isb.Message) map[string][]*isb.Message {
```

Comment:
reason I am using  `map[bufferId][]*isb.Message ` as a return type for shuffle is, writer can do parallel writes to isbs